File: Corte1/RetosEntrega19Agosto/Girar90Grados.cpp
#include <iostream>
using namespace std;

int main() {
    //Se crea y se pregunta el tamaño de la matriz (filas y columnas)
    int tamanoMatrizFilas = 0;
    cout << "Ingrese cuantas filas tiene la matriz: ";
    cin >> tamanoMatrizFilas;

    int tamanoMatrizColumnas = 0;
    cout << "Ingrese cuantas columnas tiene la matriz: ";
    cin >> tamanoMatrizColumnas;

    //Se le pregunta al usuario cada valor de la matriz
    int matriz[tamanoMatrizFilas][tamanoMatrizColumnas];
    for (int i = 0; i < tamanoMatrizFilas; i++) {
        for (int j = 0; j < tamanoMatrizColumnas; j++) {
            cout << "Ingrese el valor para la posición [" << i << "][" << j << "]: ";
            cin >> matriz[i][j];
        }
    }

    // Se gira cambiando filas por columnas y mostrando la matriz girada
    int matrizGirada[tamanoMatrizColumnas][tamanoMatrizFilas];
    for (int i = 0; i < tamanoMatrizFilas; i++) {
        for (int j = 0; j < tamanoMatrizColumnas; j++) {
            matrizGirada[j][tamanoMatrizFilas - 1 - i] = matriz[i][j];
        }
    }

    // Imprime la matriz girada 90 grados
    cout << "La matriz girada 90 grados es:" << endl;
    for (int i = 0; i < tamanoMatrizColumnas; i++) {
        for (int j = 0; j < tamanoMatrizFilas; j++) {
            cout << matrizGirada[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Corte1/RetosEntrega19Agosto/Girar90Grados.py
#Se inicializan variables
filasMatriz = 0
filasMatriz = int(input("Ingrese el número de filas de la matriz: "))
columnasMatriz = 0
columnasMatriz = int(input("Ingrese el número de columnas de la matriz: "))
matriz = []
matrizRotada = []

#For para que el usuario ingrese valores de matriz
for i in range(filasMatriz):
    fila = []
    for j in range(columnasMatriz):
        valorMatriz = int(input(f"Ingrese el valor de la posición [{i}][{j}]: "))
        fila.append(valorMatriz)
    matriz.append(fila)
#For que rota la matriz, cambia filas por columnas, la resta es para evitar que esten de primeros los valores de la primera fila porque seria 270
for i in range(columnasMatriz):
    filaRotada = []
    for j in range(filasMatriz):
        filaRotada.append(matriz[filasMatriz - 1 - j][i])
    matrizRotada.append(filaRotada)

for i in range(columnasMatriz):
    for j in range(filasMatriz):
        print(matrizRotada[i][j], end=" ")
    print()
